[#31 Use SpringBoot to launch application]

Sheptytskyid opened the pull request:
implemented Spring framework
updated project skeleton

Commit:
implemented Spring framework
updated project skeleton

holkwen commented on src/main/java/net/greatstart/controller/EventController.java:
```diff
@@ -1,4 +1,35 @@
 public class EventController {
+
+    @Autowired
+    private EventService eventService;
```
Autowired на приватных полях можно делать, конечно, но не рекомендуется. Лучше через конструктор (например). В чем смысл: при определенных настройках безопасности JVM всё поломается.

Commit:
Merge branch 'master' of https://github.com/Sheptytskyid/JavaEnterpriseFinalProjectGroup6 into spring

Commit:
implemented spring boot

Sheptytskyid commented on src/main/java/net/greatstart/model/Investment.java:
```diff
@@ -1,8 +1,11 @@
+@Component
 public class Investment {
 
     private LocalDate dateOfInvestment;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "dateOfInvestment" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Sheptytskyid commented on src/main/java/net/greatstart/model/Project.java:
```diff
@@ -1,8 +1,11 @@
 
+@Component
 public class Project {
     private boolean isVerifed;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "isVerifed" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Sheptytskyid commented on src/main/java/net/greatstart/model/Project.java:
```diff
@@ -1,8 +1,11 @@
+@Component
 public class Project {
     private boolean isVerifed;
     private String name;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "name" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Sheptytskyid commented on src/main/java/net/greatstart/dao/hibernate/EventDaoImpl.java:
```diff
@@ -2,29 +2,34 @@
 
+    @Override
     public List<Event> getAll() {
         return null;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)

Sheptytskyid commented on src/main/java/net/greatstart/model/ProjectDescription.java:
```diff
@@ -1,7 +1,10 @@
+@Component
 public class ProjectDescription {
 
     private LocalDate date;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "date" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Sheptytskyid commented on src/main/java/net/greatstart/model/User.java:
```diff
@@ -1,7 +1,10 @@
 
+@Component
 public class User {
     private int id;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "id" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Sheptytskyid commented:
SonarQube analysis reported 28 issues
* ![MAJOR][MAJOR] 28 major

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [InvestmentDaoImpl.java#L33](https://github.com/Sheptytskyid/Startup_Platform/blob/070e77825c8b532b9c1c33a5257c7e67e0d35943/src/main/java/net/greatstart/dao/hibernate/InvestmentDaoImpl.java#L33): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [ProjectDaoImpl.java#L33](https://github.com/Sheptytskyid/Startup_Platform/blob/070e77825c8b532b9c1c33a5257c7e67e0d35943/src/main/java/net/greatstart/dao/hibernate/ProjectDaoImpl.java#L33): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [UserDaoImpl.java#L38](https://github.com/Sheptytskyid/Startup_Platform/blob/070e77825c8b532b9c1c33a5257c7e67e0d35943/src/main/java/net/greatstart/dao/hibernate/UserDaoImpl.java#L38): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Investment.java#L12](https://github.com/Sheptytskyid/Startup_Platform/blob/070e77825c8b532b9c1c33a5257c7e67e0d35943/src/main/java/net/greatstart/model/Investment.java#L12): Remove this unused "project" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Investment.java#L13](https://github.com/Sheptytskyid/Startup_Platform/blob/070e77825c8b532b9c1c33a5257c7e67e0d35943/src/main/java/net/greatstart/model/Investment.java#L13): Remove this unused "inv" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Investment.java#L14](https://github.com/Sheptytskyid/Startup_Platform/blob/070e77825c8b532b9c1c33a5257c7e67e0d35943/src/main/java/net/greatstart/model/Investment.java#L14): Remove this unused "sum" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Project.java#L12](https://github.com/Sheptytskyid/Startup_Platform/blob/070e77825c8b532b9c1c33a5257c7e67e0d35943/src/main/java/net/greatstart/model/Project.java#L12): Remove this unused "desc" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Project.java#L13](https://github.com/Sheptytskyid/Startup_Platform/blob/070e77825c8b532b9c1c33a5257c7e67e0d35943/src/main/java/net/greatstart/model/Project.java#L13): Remove this unused "cost" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Project.java#L14](https://github.com/Sheptytskyid/Startup_Platform/blob/070e77825c8b532b9c1c33a5257c7e67e0d35943/src/main/java/net/greatstart/model/Project.java#L14): Remove this unused "minInvestment" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Project.java#L15](https://github.com/Sheptytskyid/Startup_Platform/blob/070e77825c8b532b9c1c33a5257c7e67e0d35943/src/main/java/net/greatstart/model/Project.java#L15): Remove this unused "addDate" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

Sheptytskyid commented on src/main/java/net/greatstart/model/User.java:
```diff
@@ -1,7 +1,10 @@
+@Component
 public class User {
     private int id;
     private String name;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "name" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Commit:
replaced xml with java-based configuration
renamed controller package to controllers
added MainController that handles root requests
removed explicit getters setters and constructors in all entities in favor of Lombok annotatins
added index.jsp simple front page

Sheptytskyid commented on src/main/java/net/greatstart/controllers/InvestmentController.java:
```diff
@@ -0,0 +1,27 @@
+package net.greatstart.controller;
```
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') This file "InvestmentController.java" should be located in "net/greatstart/controller" directory, not in "/home/travis/build/Sheptytskyid/Startup_Platform/src/main/java/net/greatstart/controllers". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1598)

Sheptytskyid commented on src/main/java/net/greatstart/config/GreatStartWebAppInitializer.java:
```diff
@@ -0,0 +1,23 @@
+//    }
+//
+//    @Override
+//    protected Class<?>[] getServletConfigClasses() {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Sheptytskyid commented on src/main/java/net/greatstart/config/GreatStartWebAppInitializer.java:
```diff
@@ -0,0 +1,23 @@
+//package ua.petshop.config;
+//
+//import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Sheptytskyid commented on src/main/java/net/greatstart/config/GreatStartWebAppInitializer.java:
```diff
@@ -0,0 +1,23 @@
+//
+//import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer;
+//
+//public class PetShopWebAppInitializer extends AbstractAnnotationConfigDispatcherServletInitializer {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Sheptytskyid commented on src/main/java/net/greatstart/config/GreatStartWebAppInitializer.java:
```diff
@@ -0,0 +1,23 @@
+//    }
+//
+//
+//}
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Sheptytskyid commented on src/main/java/net/greatstart/config/GreatStartWebAppInitializer.java:
```diff
@@ -0,0 +1,23 @@
+//public class PetShopWebAppInitializer extends AbstractAnnotationConfigDispatcherServletInitializer {
+//
+//    @Override
+//    protected String[] getServletMappings() {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Sheptytskyid commented on src/main/java/net/greatstart/config/GreatStartWebAppInitializer.java:
```diff
@@ -0,0 +1,23 @@
+//    }
+//
+//    @Override
+//    protected Class<?>[] getRootConfigClasses() {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Sheptytskyid commented on src/main/java/net/greatstart/controllers/EventController.java:
```diff
@@ -0,0 +1,35 @@
+package net.greatstart.controller;
```
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') This file "EventController.java" should be located in "net/greatstart/controller" directory, not in "/home/travis/build/Sheptytskyid/Startup_Platform/src/main/java/net/greatstart/controllers". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1598)

Sheptytskyid commented on src/main/java/net/greatstart/controllers/UserController.java:
```diff
@@ -0,0 +1,35 @@
+package net.greatstart.controller;
```
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') This file "UserController.java" should be located in "net/greatstart/controller" directory, not in "/home/travis/build/Sheptytskyid/Startup_Platform/src/main/java/net/greatstart/controllers". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1598)

Sheptytskyid commented on src/main/java/net/greatstart/config/RootConfig.java:
```diff
@@ -0,0 +1,10 @@
+//package ua.petshop.config;
+//
+//import org.springframework.context.annotation.ComponentScan;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Sheptytskyid commented on src/main/java/net/greatstart/config/RootConfig.java:
```diff
@@ -0,0 +1,10 @@
+//
+//@Configuration
+//@ComponentScan(basePackages = "ua.petshop")
+//public class RootConfig {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Sheptytskyid commented on src/main/java/net/greatstart/config/RootConfig.java:
```diff
@@ -0,0 +1,10 @@
+//@ComponentScan(basePackages = "ua.petshop")
+//public class RootConfig {
+//
+//}
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Sheptytskyid commented on src/main/java/net/greatstart/controllers/MainController.java:
```diff
@@ -0,0 +1,5 @@
+package net.greatstart.controllers;
+
+public class MainController {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this empty class, write its code or make it an "interface". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2094)

Sheptytskyid commented on src/main/java/net/greatstart/config/WebConfig.java:
```diff
@@ -0,0 +1,31 @@
+//public class WebConfig extends WebMvcConfigurerAdapter {
+//
+//    @Bean
+//    public ViewResolver viewResolver() {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Sheptytskyid commented on src/main/java/net/greatstart/config/WebConfig.java:
```diff
@@ -0,0 +1,31 @@
+//package ua.petshop.config;
+//
+//import org.springframework.context.annotation.Bean;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Sheptytskyid commented on src/main/java/net/greatstart/config/WebConfig.java:
```diff
@@ -0,0 +1,31 @@
+//    }
+//
+//    @Override
+//    public void configureDefaultServletHandling(DefaultServletHandlerConfigurer configurer) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Sheptytskyid commented on src/main/java/net/greatstart/config/WebConfig.java:
```diff
@@ -0,0 +1,31 @@
+//@Configuration
+//@EnableWebMvc
+//@ComponentScan("ua.petshop.web")
+//public class WebConfig extends WebMvcConfigurerAdapter {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Sheptytskyid commented on src/main/java/net/greatstart/config/WebConfig.java:
```diff
@@ -0,0 +1,31 @@
+//        configurer.enable();
+//    }
+//
+//}
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

Sheptytskyid commented on src/main/java/net/greatstart/controllers/ProjectController.java:
```diff
@@ -0,0 +1,35 @@
+package net.greatstart.controller;
```
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') This file "ProjectController.java" should be located in "net/greatstart/controller" directory, not in "/home/travis/build/Sheptytskyid/Startup_Platform/src/main/java/net/greatstart/controllers". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1598)

Sheptytskyid commented:
SonarQube analysis reported 26 issues
* ![CRITICAL][CRITICAL] 4 critical
* ![MAJOR][MAJOR] 18 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [InvestmentDaoImpl.java#L33](https://github.com/Sheptytskyid/Startup_Platform/blob/e64ae95da2ec4e4c95a98ab7c3991a794687969c/src/main/java/net/greatstart/dao/hibernate/InvestmentDaoImpl.java#L33): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [ProjectDaoImpl.java#L33](https://github.com/Sheptytskyid/Startup_Platform/blob/e64ae95da2ec4e4c95a98ab7c3991a794687969c/src/main/java/net/greatstart/dao/hibernate/ProjectDaoImpl.java#L33): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [UserDaoImpl.java#L38](https://github.com/Sheptytskyid/Startup_Platform/blob/e64ae95da2ec4e4c95a98ab7c3991a794687969c/src/main/java/net/greatstart/dao/hibernate/UserDaoImpl.java#L38): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MINOR][MINOR] [GreatStartWebAppInitializer.java](https://github.com/Sheptytskyid/Startup_Platform/blob/e64ae95da2ec4e4c95a98ab7c3991a794687969c/src/main/java/net/greatstart/config/GreatStartWebAppInitializer.java): Move this file to a named package. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1220)
1. ![MINOR][MINOR] [RootConfig.java](https://github.com/Sheptytskyid/Startup_Platform/blob/e64ae95da2ec4e4c95a98ab7c3991a794687969c/src/main/java/net/greatstart/config/RootConfig.java): Move this file to a named package. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1220)
1. ![MINOR][MINOR] [WebConfig.java](https://github.com/Sheptytskyid/Startup_Platform/blob/e64ae95da2ec4e4c95a98ab7c3991a794687969c/src/main/java/net/greatstart/config/WebConfig.java): Move this file to a named package. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1220)

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Commit:
replaced xml with java-based configuration
renamed controller package to controllers
added MainController that handles root requests
removed explicit getters setters and constructors in all entities in favor of Lombok annotatins
added index.jsp simple front page
turned off Travis email notifications

Sheptytskyid commented on src/main/java/net/greatstart/model/Contact.java:
```diff
@@ -1,31 +1,13 @@
 @Data
+@AllArgsConstructor
 public class Contact {
     private String address;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "address" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Sheptytskyid commented on src/main/java/net/greatstart/model/Contact.java:
```diff
@@ -1,31 +1,13 @@
+@AllArgsConstructor
 public class Contact {
     private String address;
     private String phoneNumber;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "phoneNumber" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Sheptytskyid commented on src/main/java/net/greatstart/model/User.java:
```diff
@@ -14,12 +14,4 @@
     private Contact contact;
     private List<Investment> investments;
     private List<Project> ownedProjects;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "ownedProjects" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Sheptytskyid commented on src/main/java/net/greatstart/model/User.java:
```diff
@@ -14,12 +14,4 @@
     private Contact contact;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "contact" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Sheptytskyid commented on src/main/java/net/greatstart/model/User.java:
```diff
@@ -14,12 +14,4 @@
     private Contact contact;
     private List<Investment> investments;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "investments" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Sheptytskyid commented:
SonarQube analysis reported 14 issues
* ![MAJOR][MAJOR] 14 major

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [InvestmentDaoImpl.java#L33](https://github.com/Sheptytskyid/Startup_Platform/blob/9c7565c057e1e19e1eec45a95838a4fd1567e192/src/main/java/net/greatstart/dao/hibernate/InvestmentDaoImpl.java#L33): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [ProjectDaoImpl.java#L33](https://github.com/Sheptytskyid/Startup_Platform/blob/9c7565c057e1e19e1eec45a95838a4fd1567e192/src/main/java/net/greatstart/dao/hibernate/ProjectDaoImpl.java#L33): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [UserDaoImpl.java#L38](https://github.com/Sheptytskyid/Startup_Platform/blob/9c7565c057e1e19e1eec45a95838a4fd1567e192/src/main/java/net/greatstart/dao/hibernate/UserDaoImpl.java#L38): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [User.java#L9](https://github.com/Sheptytskyid/Startup_Platform/blob/9c7565c057e1e19e1eec45a95838a4fd1567e192/src/main/java/net/greatstart/model/User.java#L9): Remove this unused "id" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L10](https://github.com/Sheptytskyid/Startup_Platform/blob/9c7565c057e1e19e1eec45a95838a4fd1567e192/src/main/java/net/greatstart/model/User.java#L10): Remove this unused "name" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L11](https://github.com/Sheptytskyid/Startup_Platform/blob/9c7565c057e1e19e1eec45a95838a4fd1567e192/src/main/java/net/greatstart/model/User.java#L11): Remove this unused "password" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L12](https://github.com/Sheptytskyid/Startup_Platform/blob/9c7565c057e1e19e1eec45a95838a4fd1567e192/src/main/java/net/greatstart/model/User.java#L12): Remove this unused "email" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L13](https://github.com/Sheptytskyid/Startup_Platform/blob/9c7565c057e1e19e1eec45a95838a4fd1567e192/src/main/java/net/greatstart/model/User.java#L13): Remove this unused "role" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

GetFire commented on src/main/java/net/greatstart/controllers/EventController.java:
```diff
@@ -0,0 +1,35 @@
+@Controller
+public class EventController {
+
+    @Autowired
```
Может стоит @Autowired вынести на конструктор?

GetFire commented on src/main/java/net/greatstart/controllers/InvestmentController.java:
```diff
@@ -0,0 +1,27 @@
+public class InvestmentController {
+
+    @Autowired
+    private InvestmentService investmentService;
```
Тоже @Autowired на конструктор

GetFire commented on src/main/java/net/greatstart/controllers/ProjectController.java:
```diff
@@ -0,0 +1,35 @@
+public class ProjectController {
+
+    @Autowired
+    private ProjectService projectService;
```
аналогично

GetFire commented on src/main/java/net/greatstart/controllers/UserController.java:
```diff
@@ -0,0 +1,35 @@
+
+    @Autowired
+    private UserService userService;
+
```
и тут

Commit:
added constructors to controllers and services

Sheptytskyid commented:
SonarQube analysis reported 14 issues
* ![MAJOR][MAJOR] 14 major

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [InvestmentDaoImpl.java#L33](https://github.com/Sheptytskyid/Startup_Platform/blob/c27658bc5650f4f165890307855d5c3095446144/src/main/java/net/greatstart/dao/hibernate/InvestmentDaoImpl.java#L33): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [ProjectDaoImpl.java#L33](https://github.com/Sheptytskyid/Startup_Platform/blob/c27658bc5650f4f165890307855d5c3095446144/src/main/java/net/greatstart/dao/hibernate/ProjectDaoImpl.java#L33): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [UserDaoImpl.java#L38](https://github.com/Sheptytskyid/Startup_Platform/blob/c27658bc5650f4f165890307855d5c3095446144/src/main/java/net/greatstart/dao/hibernate/UserDaoImpl.java#L38): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [User.java#L9](https://github.com/Sheptytskyid/Startup_Platform/blob/c27658bc5650f4f165890307855d5c3095446144/src/main/java/net/greatstart/model/User.java#L9): Remove this unused "id" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L10](https://github.com/Sheptytskyid/Startup_Platform/blob/c27658bc5650f4f165890307855d5c3095446144/src/main/java/net/greatstart/model/User.java#L10): Remove this unused "name" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L11](https://github.com/Sheptytskyid/Startup_Platform/blob/c27658bc5650f4f165890307855d5c3095446144/src/main/java/net/greatstart/model/User.java#L11): Remove this unused "password" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L12](https://github.com/Sheptytskyid/Startup_Platform/blob/c27658bc5650f4f165890307855d5c3095446144/src/main/java/net/greatstart/model/User.java#L12): Remove this unused "email" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L13](https://github.com/Sheptytskyid/Startup_Platform/blob/c27658bc5650f4f165890307855d5c3095446144/src/main/java/net/greatstart/model/User.java#L13): Remove this unused "role" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

zozich commented on src/main/java/net/greatstart/config/GreatStartWebAppInitializer.java:
```diff
@@ -0,0 +1,21 @@
+
+import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer;
+
+public class GreatStartWebAppInitializer extends AbstractAnnotationConfigDispatcherServletInitializer {
```
What is this?

zozich commented on src/main/java/net/greatstart/config/RootConfig.java:
```diff
@@ -0,0 +1,10 @@
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
```
I don't see SpringBoot application init anywhere

zozich reviewed (changes requested):
Please amend

Commit:
added Spring Boot launcher

Sheptytskyid commented:
SonarQube analysis reported 14 issues
* ![MAJOR][MAJOR] 14 major

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [InvestmentDaoImpl.java#L33](https://github.com/Sheptytskyid/Startup_Platform/blob/de10aba02a7b258588e1c38b9c0d50eff99dd754/src/main/java/net/greatstart/dao/hibernate/InvestmentDaoImpl.java#L33): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [ProjectDaoImpl.java#L33](https://github.com/Sheptytskyid/Startup_Platform/blob/de10aba02a7b258588e1c38b9c0d50eff99dd754/src/main/java/net/greatstart/dao/hibernate/ProjectDaoImpl.java#L33): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [UserDaoImpl.java#L38](https://github.com/Sheptytskyid/Startup_Platform/blob/de10aba02a7b258588e1c38b9c0d50eff99dd754/src/main/java/net/greatstart/dao/hibernate/UserDaoImpl.java#L38): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [User.java#L9](https://github.com/Sheptytskyid/Startup_Platform/blob/de10aba02a7b258588e1c38b9c0d50eff99dd754/src/main/java/net/greatstart/model/User.java#L9): Remove this unused "id" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L10](https://github.com/Sheptytskyid/Startup_Platform/blob/de10aba02a7b258588e1c38b9c0d50eff99dd754/src/main/java/net/greatstart/model/User.java#L10): Remove this unused "name" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L11](https://github.com/Sheptytskyid/Startup_Platform/blob/de10aba02a7b258588e1c38b9c0d50eff99dd754/src/main/java/net/greatstart/model/User.java#L11): Remove this unused "password" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L12](https://github.com/Sheptytskyid/Startup_Platform/blob/de10aba02a7b258588e1c38b9c0d50eff99dd754/src/main/java/net/greatstart/model/User.java#L12): Remove this unused "email" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L13](https://github.com/Sheptytskyid/Startup_Platform/blob/de10aba02a7b258588e1c38b9c0d50eff99dd754/src/main/java/net/greatstart/model/User.java#L13): Remove this unused "role" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

Commit:
added jsp compiler dependency

Sheptytskyid commented:
SonarQube analysis reported 14 issues
* ![MAJOR][MAJOR] 14 major

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [InvestmentDaoImpl.java#L33](https://github.com/Sheptytskyid/Startup_Platform/blob/de7a5929c0f614cfa483932a680ebfb710e1de96/src/main/java/net/greatstart/dao/hibernate/InvestmentDaoImpl.java#L33): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [ProjectDaoImpl.java#L33](https://github.com/Sheptytskyid/Startup_Platform/blob/de7a5929c0f614cfa483932a680ebfb710e1de96/src/main/java/net/greatstart/dao/hibernate/ProjectDaoImpl.java#L33): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [UserDaoImpl.java#L38](https://github.com/Sheptytskyid/Startup_Platform/blob/de7a5929c0f614cfa483932a680ebfb710e1de96/src/main/java/net/greatstart/dao/hibernate/UserDaoImpl.java#L38): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [User.java#L9](https://github.com/Sheptytskyid/Startup_Platform/blob/de7a5929c0f614cfa483932a680ebfb710e1de96/src/main/java/net/greatstart/model/User.java#L9): Remove this unused "id" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L10](https://github.com/Sheptytskyid/Startup_Platform/blob/de7a5929c0f614cfa483932a680ebfb710e1de96/src/main/java/net/greatstart/model/User.java#L10): Remove this unused "name" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L11](https://github.com/Sheptytskyid/Startup_Platform/blob/de7a5929c0f614cfa483932a680ebfb710e1de96/src/main/java/net/greatstart/model/User.java#L11): Remove this unused "password" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L12](https://github.com/Sheptytskyid/Startup_Platform/blob/de7a5929c0f614cfa483932a680ebfb710e1de96/src/main/java/net/greatstart/model/User.java#L12): Remove this unused "email" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [User.java#L13](https://github.com/Sheptytskyid/Startup_Platform/blob/de7a5929c0f614cfa483932a680ebfb710e1de96/src/main/java/net/greatstart/model/User.java#L13): Remove this unused "role" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

zozich reviewed (approved):
Ok now
